data_write/data_write.py

```diff
@@ -466,6 +556,54 @@ def rawrf_locations(self):
         """
         return self._rawrf_locations
 
+    @property
+    def rawrf_num_samps(self):
+        """ Gets the number of rawrf samples per antenna.
+
+        Returns:
+            TYPE: int
+        """
+        return self._rawrf_num_samps
+
+    @property
+    def gps_locked(self):
+        """ Return the boolean value indicating if the GPS was locked during the entire int period
+
+        Returns:
+            TYPE: Boolean.
+        """
+        return self._gps_locked
+
+    @property
+    def gps_to_system_time_diff(self):
+        """ Gets the maximum time diff in seconds between the GPS (box_time) and system (NTP) during
+        the integration period. Negative if GPS time is ahead of system/NTP time.
+
+        Returns:
+            TYPE: Double.
+        """
+        return self._gps_to_system_time_diff
+
+    @property
+    def agc_status_word(self):
+        """
+        AGC Status, a '1' in bit position corresponds to an AGC fault on that transmitter
+
+        Returns:
+             TYPE: Int
+        """
+        return self._agc_status_word
+
+    @property
+    def lp_status_word(self):
+        """
+        Low Power, a '1' in bit position corresponds to a low power condition on that transmitter
+
+        Returns:
+             TYPE: Int
+        """
+        return self._lp_status_word
+
 
 class DataWrite(object):
```

Comment:
I think it's passed as a way to avoid re-parsing the config file, since it's already completely loaded into memory. It would be better to change the ParseData __init__ (and line 267) so that they operate this way as well. Avoiding I/O is preferable in this case, in my opinion.

Reply:
I implemented this, it seems to be working just fine.